### DISCUSSION TITLE
[closed][x][validation] How to validate term_value to the term_iri namespace systematically?

### DISCUSSION THREAD

**Discussion by sveinugu** [fairtracks/fairtracks_standard#80] [Team Posts]
Should we add something to the JSON schema to denote these relationships?

**Comment by sveinugu** [comment]
José María's answer: We could add a check, and at most, we should fire some warning, but not complaining. In case of doubt, term_iri might be the reliable source.

**Comment by sveinugu** [comment]
@jmfernandez But the JSON schema itself do not contain anything about this relationship, other than that the attributes are located together. Should we formalize this relationship, and if so how?

**Comment by jmfernandez** [comment]
Good question @sveinugu , as that is a higher level of check, as it requires accessing both values.

**Comment by sveinugu** [comment]
@jmfernandez Old issue. What is the current status of the validator? Do you validate the term_value terms? If so, is the connection between the term_url fields and the term_value fields just hardcoded? Should we not formalize this relation?

**Comment by jmfernandez** [comment]
No validation is currently being done on these pairs. We have to keep in mind this, so we include the needed hints in the JSON Schema after the needed validator extensions are implemented (and its acknowledged syntax, defined).

**Comment by jmfernandez** [comment]
Issue https://github.com/fairtracks/fairtracks_validator/issues/3

**Comment by sveinugu** [comment]
@jmfernandez I suppose a simple solution would be to add an attribute to the term_value fields that contains a json reference to the term_url field. The name of this attribute could be e.g. “term_url_ref”  I assume it makes most sense for the validation service that the reference goes from term_value to term_url, as it is term_value that needs to be validated. 

For completeness, we should probably also define an attribute “primary_label” or similar that will be set to true for, I think, all our “term_value” fields. 

Btw: perhaps term_label is a better name than term_value throughout?

**Comment by sveinugu** [comment]
Have changed term_url -> term_id and term_value -> term_label to follow standard ontology naming, in: be8887a0236b07a23de42ee01769266b1f828587.

There is no point in defining a "primary_label". I think it was supposed to enforce that only the primary label was allowed in the field. But I think we always should enforce this anyway, so ignoring my own suggestion.

Have created new issue https://github.com/fairtracks/fairtracks_standard/issues/36, so closing this.
